wrc/codegen/merger.py

```diff
@@ -0,0 +1,177 @@
+        super().__init__(message)
+
+
+def compare_article_numbers(num1: str, num2: str) -> bool:
+    """
+    Returns True if num1 < num2, otherwise returns False.
+    This is necessary because:
+    * We need to compare article numbers like '1' and 'A'.
+    * '2' > '10' == True, which is a problem.
+    """
+    is_digit_1 = num1.isdigit()
+    is_digit_2 = num2.isdigit()
+    if is_digit_1 and is_digit_2:
+        res = int(num1) < int(num2)
+    elif is_digit_1 and not is_digit_2:
+        # Only num2 is a letter.
+        res = True
+    elif not is_digit_1 and not is_digit_2:
+        res = num1 < num2
+    else:
+        res = False
+
+    return res
+
+
+def compare_reg_numbers(num1: list, num2: list) -> bool:
+    """
+    Returns True if num1 < num2, otherwise returns False.
+    We need this for the same reasons we need compare_article_numbers().
+    """
+    for n1, n2 in zip(num1, num2):
+        if n1 < n2:
+            return True
+        elif n1 > n2:
+            return False
+
+    return len(num1) < len(num2)
+
+
+def get_reg_number(reg):
+    if isinstance(reg, Guideline):
+        reg = reg.number.split('+')[0]
+        num = split_rule_number(reg)
+    else:
+        num = split_rule_number(reg.number)
+    if str(num[-1]) == '0':
+        # Because split_rule_number() sometimes returns a 0 at the end, and we don't want that here.
+        del num[-1]
+    return num
+
+
+def merge_ast(astreg, astguide, language_options):
+    """
+    Combine the two AST into one.
+    """
+    # We only want the Articles, so we will start the for loop below at FIRST_ARTICLE_INDEX,
+    # and check if the indices for Notes, TOC and First Article are correct.
+    if not isinstance(astreg.sections[NOTES_INDEX], Section) or \
+            not isinstance(astguide.sections[NOTES_INDEX], Section):
+        raise BadFormatError('NOTES_INDEX is incorrect')
+    if not isinstance(astreg.sections[TOC_INDEX], TableOfContent):
+        raise BadFormatError('TOC_INDEX is incorrect')
+    if not (hasattr(astreg.sections[FIRST_ARTICLE_INDEX], 'number') and astreg.sections[
+                FIRST_ARTICLE_INDEX].number == '1'):
+        raise BadFormatError('FIRST_ARTICLE_INDEX is incorrect.')
+
+    # We take the astreg as the base, and then we add the guidelines in their corresponding places.
+    ast_combined = astreg
+
+    # Merge the notes:
+    # We won't include the "WCA Regulations" notes from the Guidelines, as it is the same as in
+    # "WCA Regulations and Guidelines" of the Regulations document.
+    # languages_options is used to handle translations.
+    ast_combined.sections[NOTES_INDEX].content.append(
+        [cont for cont in astguide.sections[NOTES_INDEX].content if language_options['regulations'] not in cont.title])
+
+    # Now we iterate over all the Articles found in astguide, starting from FIRST_ARTICLE_INDEX.
+    for guide_section in astguide.sections[FIRST_ARTICLE_INDEX:]:
```

Comment:
Might be worth storing that in its own (temporary) variable, otherwise it's easy to oversee (or get confused about) the `:` at the end of the array brackets.

Reply:
Changed to
```python3
# Now we iterate over all the Articles found in astguide, from FIRST_ARTICLE_INDEX until the end of the
# sections list.
guide_sections = astguide.sections[FIRST_ARTICLE_INDEX:]
for guide_section in guide_sections:
```
Looks better?